server.py

```diff
@@ -89,25 +124,35 @@ def run_server():
                 msg_recv = conn.recv(4096)
                 msg += msg_recv
                 if len(msg_recv) < 4096:
-                    # import pdb; pdb.set_trace()
                     try:
-                        client_response = parse_request(msg)
+                        parsed_response = parse_request(msg)
+                        body, content_type = resolve_uri(parsed_response)
+                        server_response = response_ok(body, content_type)
                     except NotImplementedError:
-                        client_response = response_error(
-                            b"HTTP/1.1 405 Method Not Allowed\r\n",
-                            b"GET method required.\r\n"
+                        server_response = response_error(
+                            b"405",
+                            b"Method Not Allowed",
+                            b"GET method required."
                         )
                     except NameError:
-                        client_response = response_error(
-                            b"HTTP/1.1 400 Bad Request\r\n",
-                            b"Not a valid HTTP/1.1 request.\r\n"
+                        server_response = response_error(
+                            b"400",
+                            b"Bad Request",
+                            b"Not a valid HTTP/1.1 request."
                         )
                     except ValueError:
-                        client_response = response_error(
-                            b"HTTP/1.1 406 Not Acceptable\r\n",
-                            b"'Host' header required.\r\n"
+                        server_response = response_error(
+                            b"406",
+                            b"Not Acceptable",
+                            b"'Host' header required."
                         )
-                    conn.sendall(client_response)
+                    except OSError:
+                        server_response = response_error(
+                            b"420",
+                            b"Enhance Your Calm",
+                            b"Keyboard Driver Error"
+                        )
```

Comment:
I enjoy the joke here, but is this really the "correct" response code to return?  I think there are better choices.